[Bug]: Linting fails on v8.0.0 javascript projects

### What's not working?

`yarn rw lint` fails with the following error:

```
/Users/jgmw/Development/redwood/rw-test/v8-javascript/api/src/lib/auth.js
  23:31  error  'roles' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars

/Users/jgmw/Development/redwood/rw-test/v8-javascript/web/src/App.jsx
  9:2  error  Parsing error: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (9:2)

/Users/jgmw/Development/redwood/rw-test/v8-javascript/web/src/Routes.jsx
  14:4  error  Parsing error: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (14:4)

/Users/jgmw/Development/redwood/rw-test/v8-javascript/web/src/entry.client.jsx
  24:4  error  Parsing error: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (24:4)

/Users/jgmw/Development/redwood/rw-test/v8-javascript/web/src/pages/FatalErrorPage/FatalErrorPage.jsx
  15:4  error  Parsing error: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (15:4)

/Users/jgmw/Development/redwood/rw-test/v8-javascript/web/src/pages/NotFoundPage/NotFoundPage.jsx
  2:2  error  Parsing error: This experimental syntax requires enabling one of the following parser plugin(s): "jsx", "flow", "typescript". (2:2)

✖ 6 problems (6 errors, 0 warnings)
```

This occurs for the default redwood template when you select javascript over typescript.

### How do we reproduce the bug?

* `yarn create redwood-app` 
* select javascript 
* `yarn rw lint`

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 14.6.1
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.16.0 - /private/var/folders/zb/kc7f0z2d6973pg1p0pjz5nhm0000gn/T/xfs-ba1c093d/node
    Yarn: 4.4.0 - /private/var/folders/zb/kc7f0z2d6973pg1p0pjz5nhm0000gn/T/xfs-ba1c093d/yarn
  Databases:
    SQLite: 3.43.2 - /usr/bin/sqlite3
  Browsers:
    Chrome: 128.0.6613.120
    Safari: 17.6
  npmPackages:
    @redwoodjs/core: 8.0.0 => 8.0.0 
    @redwoodjs/project-config: 8.0.0 => 8.0.0 
  redwood.toml:
    [web]
      title = "Redwood App"
      port = 8910
      apiUrl = "/.redwood/functions" # You can customize graphql and dbauth urls individually too: see https://redwoodjs.com/docs/app-configuration-redwood-toml#api-paths
      includeEnvironmentVariables = [
        # Add any ENV vars that should be available to the web side to this array
        # See https://redwoodjs.com/docs/environment-variables#web
      ]
    [api]
      port = 8911
    [browser]
      open = true
    [notifications]
      versionUpdates = ["latest"]
```


### Are you interested in working on this?

- [X] I'm interested in working on this

--- a/.changesets/11458.md
+++ b/.changesets/11458.md
@@ -1,0 +1,3 @@
+- fix(linting): Re-enable babel presets during linting of javascript projects (#11458) by @Josh-Walker-GM
+
+The `yarn rw lint` command was failing for JavaScript projects. This change re-enables certain babel plugins to correct this issue and allow this command to succeed again.

--- a/packages/babel-config/src/web.ts
+++ b/packages/babel-config/src/web.ts
@@ -20,6 +20,7 @@ export interface Flags {
   forJest?: boolean // will change the alias for module-resolver plugin
   forPrerender?: boolean // changes what babel-plugin-redwood-routes-auto-loader does
   forRsc?: boolean
+  forJavaScriptLinting?: boolean // will enable presets to supporting linting in the absence of typescript related presets/plugins/parsers
 }
 
 export const getWebSideBabelPlugins = (
@@ -166,7 +167,7 @@ export const getWebSideOverrides = (
 export const getWebSideBabelPresets = (options: Flags) => {
   // When we perform prerendering we don't use vite, so we need to add the
   // appropriate presets for react, env, and typescript, etc.
-  if (options.forPrerender || options.forJest) {
+  if (options.forPrerender || options.forJest || options.forJavaScriptLinting) {
     let reactPresetConfig: babel.PluginItem = { runtime: 'automatic' }
 
     // This is a special case, where @babel/preset-react needs config

--- a/packages/eslint-config/index.js
+++ b/packages/eslint-config/index.js
@@ -7,7 +7,7 @@ const {
   getApiSideDefaultBabelConfig,
   getWebSideDefaultBabelConfig,
 } = require('@redwoodjs/babel-config')
-const { getConfig } = require('@redwoodjs/project-config')
+const { getConfig, isTypeScriptProject } = require('@redwoodjs/project-config')
 
 const config = getConfig()
 
@@ -16,7 +16,10 @@ const getProjectBabelOptions = () => {
   // So we just take it out and put it as a separate item
   // Ignoring overrides, as I don't think it has any impact on linting
   const { overrides: _webOverrides, ...otherWebConfig } =
-    getWebSideDefaultBabelConfig()
+    getWebSideDefaultBabelConfig({
+      // We have to enable certain presets like `@babel/preset-react` for JavaScript projects
+      forJavaScriptLinting: !isTypeScriptProject(),
+    })
 
   const { overrides: _apiOverrides, ...otherApiConfig } =
     getApiSideDefaultBabelConfig()

--- a/packages/project-config/src/paths.ts
+++ b/packages/project-config/src/paths.ts
@@ -440,3 +440,11 @@ export function projectIsEsm() {
 
   return true
 }
+
+export const isTypeScriptProject = () => {
+  const paths = getPaths()
+  return (
+    fs.existsSync(path.join(paths.web.base, 'tsconfig.json')) ||
+    fs.existsSync(path.join(paths.api.base, 'tsconfig.json'))
+  )
+}

